src/node_buffer.cc

```diff
@@ -701,6 +928,50 @@ void IndexOfNumber(const FunctionCallbackInfo<Value>& args) {
 }
 
 
+void IsNull(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!HasInstance(args[0]))
+    return env->ThrowTypeError("first arg should be a Buffer");
+
+  ARGS_THIS(args[0].As<Object>())
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset, obj_length);
+
+  char *ptr = obj_data + offset;
+
+  args.GetReturnValue().Set( ptr == NULL );
+}
+
+
+void Address(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+
+  if (!HasInstance(args[0]))
+    return env->ThrowTypeError("first arg should be a Buffer");
+
+  ARGS_THIS(args[0].As<Object>())
+
+  uint32_t offset = args[1]->Uint32Value();
+  CHECK_LE(offset, obj_length);
+
+  char *ptr = obj_data + offset;
+
+  // pointer-size * 2 (for hex printout) + 1 null byte
+  char strbuf[(sizeof(char *) * 2) + 1];
+
+  // using stringstream and std::hex here instead of printf(%#) because
+  // of cross-platform differences experienced with the later
+  std::stringstream stream;
+  stream.rdbuf()->pubsetbuf(strbuf, sizeof(strbuf));
+
+  stream << std::hex << (uintptr_t)ptr << '\0';
```

Comment:
> src/node_buffer.cc:17:  Streams are highly discouraged.  [readability/streams] [3]

Do you have a better suggestion? I don't really know another way of manually using `std::hex`.
`printf("%p")` has given me varying results cross-platform in the past (upper-lower casing, `0x` prefix there or not).
Ideally we're returning a value that's cross-platform here IMO.